File: keyphrase-extraction/amazon-comprehend/AmazonComprehendKeyphraseExtraction.java
// For English, set "languageCode" to "en".

AmazonComprehend client = AmazonComprehendClientBuilder
    .standard()
    .withCredentials(credentialsProvider)
    .withRegion(Regions.US_EAST_1)
    .build();

DetectKeyPhrasesRequest request = new DetectKeyPhrasesRequest()
    .withText(text)
    .withLanguageCode(languageCode);

DetectKeyPhrasesResult result = client.detectKeyPhrases(request);

List<KeyPhrase> keyPhrases = result.getKeyPhrases();

File: keyphrase-extraction/amazon-comprehend/AmazonComprehendKeyphraseExtraction.kt
// "text" is a string containing the input document.
// For English, set "languageCode" to "en".

val client = AmazonComprehendClientBuilder
    .standard()
    .withCredentials(credentialsProvider)
    .withRegion(Regions.US_EAST_1)
    .build()

val request = DetectKeyPhrasesRequest()
    .withText(text)
    .withLanguageCode(languageCode)

val result = client.detectKeyPhrases(request)

val keyPhrases = result.keyPhrases
